File: go/config/Ussd.go
package config

import (
	"github.com/lvcdy/huawei-lte-api-go/session"
)

// Ussd 对应 config/Ussd.py。
type Ussd struct {
	*session.ApiGroup
}

// NewUssd 创建 Ussd API 分组。
func NewUssd(s *session.Session) *Ussd {
	return &Ussd{ApiGroup: session.NewApiGroup(s)}
}

// Prepaidussd 对应 ussd/prepaidussd.xml。
func (u *Ussd) Prepaidussd() (map[string]interface{}, error) {
	return u.S.Get("ussd/prepaidussd.xml", nil, "config")
}

// Postpaidussd 对应 ussd/postpaidussd.xml。
func (u *Ussd) Postpaidussd() (map[string]interface{}, error) {
	return u.S.Get("ussd/postpaidussd.xml", nil, "config")
}

File: src/config/ussd.rs
//! Ussd config group (`config/Ussd.py`).

use serde_json::Value;

use crate::api::get_value;
use crate::connection::Connection;
use crate::errors::Result;

/// Ussd config group.
pub struct UssdConfig<'a> {
    conn: &'a Connection,
}

impl<'a> UssdConfig<'a> {
    /// Build a group bound to `conn`.
    pub fn new(conn: &'a Connection) -> Self {
        UssdConfig { conn }
    }

    /// `ussd/prepaidussd.xml`. Prepaid USSD configuration.
    pub fn prepaidussd(&self) -> Result<Value> {
        get_value(self.conn, "config/ussd/prepaidussd.xml")
    }

    /// `ussd/postpaidussd.xml`. Postpaid USSD configuration.
    pub fn postpaidussd(&self) -> Result<Value> {
        get_value(self.conn, "config/ussd/postpaidussd.xml")
    }
}
